tests/unit/test_logging.py

```diff
@@ -248,15 +248,26 @@ def test_root_logger_configuration(root_logger_reset):  # noqa: F811
     assert any(isinstance(handler, RecordCompiler) for handler in root_handlers)
 
 
-def test_secrets_logging_config(root_logger_reset, expect_json_log):  # noqa: F811
-    """Make sure secret values are not revealed when logging the configuration."""
+def test_secrets_logging_config(root_logger_reset, capsys):  # noqa: F811
+    """Make sure secret values are not revealed when logging the configuration.
 
-    class TestConfig(LoggingConfig):
-        secret_bytes: SecretBytes = Field(default=b"test")
-        secret_str: SecretStr = Field(default="test")
-
-    with expect_json_log() as json_log:
-        configure_logging(config=TestConfig(service_name="", service_instance_id=""))
+    Additionally, make sure all other values *are* logged.
+    """
 
-    assert json_log.details["secret_bytes"] == "**********"
-    assert json_log.details["secret_str"] == "**********"
+    class TestConfig(LoggingConfig):
+        secret_bytes: SecretBytes = Field(default=b"my secret bytes")
+        secret_str: SecretStr = Field(default="my secret str")
+
+    config = TestConfig(service_name="", service_instance_id="")
+
+    configure_logging(config=config)
+    out, err = capsys.readouterr()
+    printed_log = out + err
+
+    # Verify that secret values are not contained in log, but everything else is.
+    for key, val in config.model_dump().items():
+        assert key in printed_log
+        if isinstance(val, (SecretBytes, SecretStr)):
+            assert str(val.get_secret_value()) not in printed_log
+        elif val:
+            assert val in printed_log
```

Comment:
```suggestion
    class TestConfig(LoggingConfig):
        string: str = "foo"
        number: int = 42
        secret_bytes: SecretBytes = Field(default=b"silent")
        secret_str: SecretStr = Field(default="silent")

    config = TestConfig(service_name="", service_instance_id="")

    configure_logging(config=config)
    out, err = capsys.readouterr()
    printed_log = out + err

    assert "Logging configured, complete configuration in details" in printed_log
    assert "silent" not in printed_log
    for key, value in config.model_dump().items():
        assert key in printed_log
        if not key.startswith("secret"):
            json_value = "null" if value is None else str(value)
            assert json_value in printed_log
```

We don't need to get the secret value since we know it already. As a general rule, keep test very simple; they can contain duplication. More complicated code is more difficult to read and can contain errors, even if it looks simple. In this case, this is actually the case: `get_secret_value` would get a bytes value, and `str()` would then add a "b" and single quotes (the Python repr of the string). If "my secret str" would be contained without quotes or JSON formatted double quotes then we would not catch it. By keeping it simple, you avoid such mistakes.

As a general rule, try to make tests for exclusion unspecific, and tests for inclusion specific.

I have also added some more non-secrets that are not part of the LoggingConfig to make sure these are printed as well.

Reply:
I didn't realize the output would get formatted like that. Thanks for pointing that out.